### PR TITLE
Update Actor usage in docs to use AbstractActor, the main Java API since 2.5

### PR DESCRIPTION
**Pull request by ktoso**
Resolves https://github.com/playframework/playframework/issues/7544

### CODE REVIEW

File: framework/src/play-akka-http-server/src/test/scala/play/AkkaTestServer.scala [hunk 1]
```diff
@@ -15,7 +15,7 @@ object AkkaTestServer extends App {
     address = "127.0.0.1"
   )) {
     case GET(p"/") => Action { implicit req =>
-      Results.Ok(s"Hello world")
+      Results.Ok(s"Hello world: " + req.headers.headers.toList)
```
Comment (cchantep) [inline framework/src/play-akka-http-server/src/test/scala/play/AkkaTestServer.scala]: String interpolation?

Comment (ktoso) [inline framework/src/play-akka-http-server/src/test/scala/play/AkkaTestServer.scala]: Thanks, this was not meant to be committed actually - reverted, thanks!

**Comment by ktoso** [comment]
Addressed comment

**Comment by richdougherty** [comment]
Thanks @ktoso. Just needs a rebase to resolve the reformatting/commenting in `AkkaHttpServer`.

**Comment by ktoso** [comment]
Rebased!

**Comment by richdougherty** [comment]
I had a look at the Travis error logs. Looks like the Javadoc needs to have a `>` turned into a `&gt;`.

```
[error] /home/travis/build/playframework/playframework/framework/src/play-guice/src/main/java/play/libs/akka/AkkaGuiceSupport.java:129: error: bad use of '>'
[error]      *     .match(CreateChildActor.class, msg -> {
[error]                                                ^
```

**Comment by richdougherty** [comment]
I pushed a small fix for the Javadoc issue in another PR: #7599.

**Comment by ktoso** [comment]
Thanks 👍 
Closing, #7599 is the follow up

**Pull request closed without merging**